maven-core/src/main/java/org/apache/maven/session/scope/internal/SessionScope.java

```diff
@@ -107,7 +107,11 @@ public <T> Provider<T> scope(final Key<T> key, final Provider<T> unscoped) {
     private <T> T createProxy(Key<T> key, Provider<T> unscoped) {
         InvocationHandler dispatcher = (proxy, method, args) -> {
             method.setAccessible(true);
-            return method.invoke(getScopeState().scope(key, unscoped).get(), args);
+            try {
+                return method.invoke(getScopeState().scope(key, unscoped).get(), args);
+            } catch (InvocationTargetException e) {
+                throw e.getCause();
```

Comment:
If the target Exception is null, it will throw NPE, and this could affect exception tracing.

Reply:
the variable `e` can never be `null`.

The statement `e.getCause()` can be `null` in theory, but in practice I don't see any case which could give this.

Reply:
`e.getCause()` return `target`. I mean e.g Cause() is null, which can affect line numbers